scripts/generateRules.mjs

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env node
+
+import { readFile, writeFile } from 'fs/promises'
+import path from 'path'
+
+import 'dotenv/config'
+import chalk from 'chalk'
+import XLSX from 'xlsx'
+import lodash from 'lodash'
+
+import { DATA_SHEET_TYPES } from '../src/server/services/records.js'
+import { SERVER_DATA_DIR, EMPTY_TEMPLATE_NAME, SRC_DIR } from '../src/server/environment.js'
+
+const { merge } = lodash
+const log = (msg) => { console.log(chalk.green(msg)) }
+
+const COLNAMES = makeColNames()
+
+function makeColNames () {
+  const upcaseLetters = []
+  for (let i = 65; i <= 90; i++) {
+    upcaseLetters.push(String.fromCharCode(i))
+  }
+
+  const secondSet = upcaseLetters.map(letter => 'A' + letter)
+  const thirdSet = upcaseLetters.map(letter => 'B' + letter)
+  return [...upcaseLetters, ...secondSet, ...thirdSet]
+}
+
+function parseListVal (lvStr) {
+  if (lvStr === 'N/A') return []
+
+  let lines = lvStr.split('\n')
+
+  // remove whitespace on each pick list option
+  lines = lines.map(line => line.trim())
+
+  // un-quote each entry
+  lines = lines.map(line => line.replace(/^"/, '').replace(/"$/, ''))
+
+  return lines
+}
+
+function parseRequired (rqStr) {
+  if (rqStr === 'Required') return true
+  if (rqStr === 'Optional') return false
+  return rqStr
+}
+
+// for the given type and column, return all the EC codes where that column is shown/used
+function filterEcCodes (logic, type, columnName) {
+  const logicCodes = logic.filter(log => log.type === type)
+  if (logicCodes.length === 0) return false
+
+  const ecCodes = []
+  for (const logic of logicCodes) {
+    if (logic.columnNames[columnName]) {
+      ecCodes.push(logic.ecCode)
+    }
+  }
+
+  return ecCodes
+}
+
+async function extractRules (workbook, logic) {
+  const rules = {}
+
+  for (const sheetName of Object.keys(DATA_SHEET_TYPES)) {
+    const type = DATA_SHEET_TYPES[sheetName]
+    const sheet = workbook.Sheets[sheetName]
+
+    // entire sheet
+    const sheetRange = XLSX.utils.decode_range(sheet['!ref'])
+
+    // range A1:13
+    const headerRange = merge({}, sheetRange, {
+      s: { c: 0, r: 0 },
+      e: { r: 12 }
```

Comment:
Nit: I guess an incremental step to avoiding magic numbers would be to declare the number of header rows as a constant somewhere, and share it across callsites.